.github/workflows/release.yml

```diff
@@ -10,6 +10,9 @@ on:
       - v*.*.*
     branches:
       - main
+    paths:
+      - 'packages/opentelemetry-node/**'
+      - '.github/workflows/release.yml'
```

Comment:
reviewer note: Limit when we do a "release" run on pushes to main. It would be overkill to do so for changes to unrelated packages.